[merged] Modify

rhatdan opened the pull request.

rhatdan commented:
I would like this pull request to replace https://github.com/projectatomic/atomic/pull/379

rhatdan commented:
@mvollmer @rhvgoyal PTAL

rhatdan commented:
@rhvgoyal Can docker-storage-setup work with btrfs or aufs?

mvollmer commented on tests/unit/test_util_sh.py:
```diff
@@ -0,0 +1,74 @@
+        # Setting with extra whitespace is recognized
+        self.assertEqual(util.sh_modify_var_in_text('   VAR  =  "val"  \n', "VAR", uppercasify),
+                         'VAR="VAL"\n')
+
```
This is actually not valis bourne shell syntax.  Should we still recognize it?

mvollmer commented on Atomic/storage.py:
```diff
@@ -20,6 +20,9 @@
 
 class Storage(Atomic):
+    dss_conf = "/etc/sysconfig/docker-storage-setup"
+    dss_conf_bak = "/etc/sysconfig/docker-storage-setup" + ".bkp"
```
This is better written as `dss_conf_bak =  dss_conf + ".bkp"`.  (This was in my original code, sorry.)

rhatdan replied:
Fixed.

mvollmer commented on Atomic/storage.py:
```diff
@@ -38,6 +41,29 @@ def reset(self):
+
+    def _driver(self, driver):
+        util.sh_modify_var_in_file(self.dss_conf, "STORAGE_DRIVER",
+                                   lambda old: util.sh_set_add(old, [driver]))
```
`STORAGE_DRIVER` is not a set, so the modification function should be simply `lambda old: driver`.

mvollmer commented:
Tested manually with Cockpit, works.

mvollmer commented:
> Tested manually with Cockpit, works.

I have only tested `--add-device`.

rhvgoyal commented:
@rhatdan no, docker-storage-setup does not support btrfs or aufs graph drivers. Only devicemapper and overlay are supported.

mvollmer commented:
> I would like this pull request to replace #379

Ok.  The "add" integration test from #379 might still be interesting.

rhatdan commented:
@mvollmer @rhvgoyal PTAL

mvollmer commented:
> @mvollmer @rhvgoyal PTAL

This comment still applies, I think: 
> STORAGE_DRIVER is not a set, so the modification function should be simply `lambda old: driver`.

rhatdan commented:
@mvollmer missed that one.  Check it out now.

mvollmer commented on Atomic/storage.py:
```diff
@@ -38,6 +41,29 @@ def reset(self):
+            if util.call(["docker-storage-setup"]) != 0:
+                os.rename(self.dss_conf_bak, self.dss_conf)
+                util.call(["docker-storage-setup"])
+                raise ValueError("Not all devices could be added")
```
The error message only makes sense for `--add-device`.  We might not need a message at all since dpcker-storage-setup should have explained itself already.

mvollmer commented:
Tested --driver manually as well.  Works as expected, except for the error message.

rhatdan commented:
Changed the error message to just show that docker-storage-setup failed, also added an exception block to bring back /etc/sysconfig/docker-storage-setup.

mvollmer commented:
LGTM, thanks!

rhatdan commented:
@rh-atomic-bot r+
